src/dotnet/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -37,7 +45,19 @@ public WattTimeDataSource(ILogger<WattTimeDataSource> logger, IWattTimeClient cl
     }
 
     /// <inheritdoc />
-    public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(Location location, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime)
+    public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime)
+    {
+        this.Logger.LogInformation("Getting carbon intensity for locations {locations} for period {periodStartTime} to {periodEndTime}.", locations, periodStartTime, periodEndTime);
+        List<EmissionsData> result = new ();
+        foreach (var location in locations)
+        {
+            IEnumerable<EmissionsData> interimResult = await GetCarbonIntensityAsync(location, periodStartTime, periodEndTime);
```

Comment:
Not blocking for this PR, but we should have an opinion on where to start telemetry activities.  My preference is in the public method, but there's no existing pattern yet, so we likley need to decide, document, and cleanup across the board

Reply:
Yep, see my comment above. Removed for now and think we should kick to a telemetry/monitoring PR